proc/process.c

```diff
@@ -1092,12 +1093,14 @@ static void process_exec(thread_t *current, process_spawn_t *spawn)
 	if (err == 0) {
 		stack = process_putargs(stack, &spawn->envp, &count);
 		stack = process_putargs(stack, &spawn->argv, &count);
-
-		/* temporary? put arguments to main on stack */
-		PUTONSTACK(stack, char **, spawn->envp);
-		PUTONSTACK(stack, char **, spawn->argv);
-		PUTONSTACK(stack, int, count);
-		PUTONSTACK(stack, void *, NULL); /* return address */
+		/* clang-format off */
+		hal_stackPutArgs(&stack, 4, (const struct stackArg[]) {
```

Comment:
Let's keep kernel somewhat conservative and use intermediate const var for this struct ;)